segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/SegmentAggregator.java

```diff
@@ -1176,6 +1176,9 @@ private long getRolloverSize() {
         // Configured value.
         long rolloverSize = this.metadata.getAttributes().getOrDefault(Attributes.ROLLOVER_SIZE, SegmentRollingPolicy.NO_ROLLING.getMaxLength());
 
+        // rolloverSize being zero means the default value should be used.
+        rolloverSize = rolloverSize == 0 ? SegmentRollingPolicy.NO_ROLLING.getMaxLength() : rolloverSize;
```

Comment:
Why would this ever be 0? Is it because it's now user-supplied? If so, what if it's negative?

Reply:
Thanks for the review. 
Yes, I remember you mentioned in the issue that if this field is zero then it means application doesn't care and a default value should be used. So it is possible that a zero is passed down to here. 
I have set validation logic in the `PravegaRequestProcessor` to ensure that only non negative value can be used.